### PR TITLE
General cleanup

### PR DESCRIPTION
**Pull request by jpouellet**
I'm actually going to use a different library because of the license ambiguities here, but I figured I'd send my changes back anyway in case you wanted them.

I have:
 - Moved most stuff out of the header, and made static everything that shouldn't export a symbol. This avoids global namespace pollution and makes it somewhat more suitable for integration with other work.
 - Rewrote the example API consumer program
   * Now uses ANSI terminal escapes instead of unicode characters when rendering to the terminal.
   * New usage: `qr_encode [-l level] [-v version] [-o file] [-m margin] <input string>`
 - Improved documentation with clearer API definition and more complete example.

I have not:
 - (Intentionally) changed any of the actual QR logic.

I would also recommend that you merge one of the other commits which fixes the ZeroMemory() sizeof() issues. It is still a real problem.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,11 @@
-qr_encode: main.c QR_Encode.c
-	rm -f qr_encode
-	gcc -std=c99 main.c QR_Encode.c -o qr_encode
+PROG = qr_encode
+OBJS = main.o QR_Encode.o
+CFLAGS = -std=c99 -D_XOPEN_SOURCE
```
Comment (swex) [inline Makefile]: why do we need it?

**Comment by swex** [comment]
Hi. I basically like your rewrite.
1. But please fix all gcc warnings.
2. Why do we need any CFLAGS here?

**Comment by jpouellet** [comment]
1. There are no gcc warnings from the part I wrote, they are all from the existing code that I did not touch (except to make things static to avoid symbol pollution).
```
$ gcc -Wall -Wextra -c main.c 2>&1 | wc -l     
0
$ gcc -Wall -Wextra -c QR_Encode.c 2>&1 | wc -l
168
```
Some of the warnings in QR_Encode.c are *actual bugs*, most notably the fact that the ZeroMemory() macro uses sizeof() on pointers expecting to get the length of the actual buffers they point to (leading to only zeroing the first 4 or 8 bytes of the much longer buffer). Other people have produced fixes for these issues and I suggest they be merged (in a separate pull request, since they're completely unrelated to these changes).

2. I believe they could be removed. I kept the `-std=c99` because it was there before, and added the `-D_XOPEN_SOURCE` because at one point my compiler was complaining about getopt() without it, but it's not necessary. Would you like me to remove them for this PR?

**Comment by swex** [comment]
Yep you are right. I'll add zero memory fixes by hands soon. thanks for pull request

**Comment by swex** [comment]
also what library are you going to use?

**Pull request merged**